query/src/main/java/org/infinispan/query/clustered/ClusteredQueryInvoker.java

```diff
@@ -101,9 +72,8 @@ QueryResponse unicast(Address address, SegmentsClusteredQueryCommand cmd) {
 
       List<QueryResponse> results = new ArrayList<>();
       try {
-         results.add(localResponse.get());
-         List<QueryResponse> responseList = sequence(futureRemoteResponses).get();
-         results.addAll(responseList);
+         results.add(await(localResponse.toCompletableFuture()));
```

Comment:
I worry that the blocking is now just relegated to this call instead. We really need https://issues.redhat.com/browse/ISPN-11731 to verify these are all fixed.

Reply:
agreed, but this method is called by the REST server when doing queries, shouldn't Blockhound bark?

Reply:
My guess is that the invoker is already on the blocking thread. But I am not certain what code path this is.